code/src/java/pcgen/gui3/JFXPanelFromResource.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 (C) Eitan Adler <lists@eitanadler.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.     See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+package pcgen.gui3;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import pcgen.util.Logging;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+
+/**
+ * Displays HTML content as a "panel".
+ */
+public final class JFXPanelFromResource<T> extends JFXPanel
+{
+
+	private final FXMLLoader fxmlLoader = new FXMLLoader();
+
+	public JFXPanelFromResource(Class<T> klass, String resourceName)
```

Comment:
How permanent are these classes?  How much Javadoc should we be doing for them as a result of the first question?

Reply:
Fairly permanent. They'd only go away once 90-100% of the code is converted. Will add docs.